### ISSUE TITLE
Twitter preview not showing `fit` tranforms correctly

### ISSUE THREAD

**Issue by bossanova808** [nystudio107/craft-seomatic#358]
Ok, so I have a portrait orientation image, and with both Twitter and Facebook set to fit, which should (according to Craft docs):

```
Fit – Scales the image so that it is as big as possible with all dimensions fitting within the specified width and height.
```

Then I would expect the preview image to show the image completely fit inside the box with whitespace around the left and right edges.  However this is what shows:

![image](https://user-images.githubusercontent.com/731309/55047498-1db30280-5099-11e9-8475-32f0b1913579.png)

..the Facebook image is as expected, but the Twitter image is shown incorrectly as cropped, though.

I have not dug into how this all works, but I do note that if I change the css property for
```
.TwitterCard-image-wrapper {
   ...
    background-size: cover
}
```
to
```
.TwitterCard-image-wrapper {
   ...
    background-size: contain
}
```

Then the display is as I'd expect (except maybe the background tone should be white and not grey?):

![image](https://user-images.githubusercontent.com/731309/55047625-81d5c680-5099-11e9-94e0-88c780e77853.png)

(More than likely this is just an issue with how the preview shows, and the actual image served to Twitter is correctly scaled, but I don't know for sure as this isn't live yet...)

**Comment by khalwat** [comment]
Can you tell me if it is just the preview that appears to be wrong, or the actual tweeted image that appears to be wrong?

**Comment by bossanova808** [comment]
Not really until I launch this version (~ a month from now) - and actually tweet something out...but no probs coming back to it then if that works for you?

**Comment by khalwat** [comment]
This works using Craft's concept of "fit" which may not be what we intuitively think it should be. But sure, lmk

**Comment by bossanova808** [comment]
It's the actual tweet too:

![image](https://user-images.githubusercontent.com/731309/59477143-3c09ec80-8e97-11e9-9ed1-0c511d3c0d2c.png)

(from https://imagescience.com.au/products/calibrators/x-rite-i1display-pro)

**Comment by lindseydiloreto** [comment]
I'm having a similar issue. It seems as if the "SEO Image Transform Mode" field is having no effect.

<img width="707" alt="Ω 2019-06-25 at 4 30 12 PM" src="https://user-images.githubusercontent.com/5309692/60140408-8bd59700-9766-11e9-89a4-da2b56f77d72.png">

It seems like whichever option I choose, the image just gets cropped...

**Twitter:**
<img width="595" alt="Ω 2019-06-25 at 4 27 48 PM" src="https://user-images.githubusercontent.com/5309692/60140416-91cb7800-9766-11e9-86ac-74e3da475430.png">

**Discord:**
<img width="457" alt="Ω 2019-06-25 at 4 25 05 PM" src="https://user-images.githubusercontent.com/5309692/60140425-9728c280-9766-11e9-95d7-0fea586ccb8b.png">

**Slack:**
<img width="598" alt="Ω 2019-06-25 at 4 31 28 PM" src="https://user-images.githubusercontent.com/5309692/60140460-c2131680-9766-11e9-9282-e84637095332.png">

Is this a bug? Or should I be selecting a different option?

---

SEOmatic: 3.2.9
Craft: 3.1.31

**Comment by lindseydiloreto** [comment]
Looking at the code being generated by SEOmatic, it seems like the "Fit" selection is being ignored by the meta tags. Both of these tags are apparently using the `crop` method.

```html
<meta content=".../_1200x630_crop_center-center_82_none/..." property="og:image">
<meta name="twitter:image" content=".../_800x418_crop_center-center_82_none/...">
```

Although oddly, the JSON seems to be respecting the `fit` method...

```js
"image":{"@type":"ImageObject","url":".../_1200x630_fit_center-center_82_none/..."}
```

**Comment by khalwat** [comment]
So probably you need to set it to `fit` for Twitter and Facebook too @lindseydiloreto ?

As for it affecting the tweet as well as the preview... Craft's idea of "fit" is that both dimensions of the image fit into what you pass in.

Twitter will crop anything passed in that is not the exact proportions that it's looking for, so the preview is accurate.

What you'd need to do what you think you want would be something that made an image of the exact dimensions passed in, and letterboxed your actual image.

**Comment by lindseydiloreto** [comment]
The Twitter and Facebook ones just say "Same As SEO Image":

<img width="657" alt="Ω 2019-06-26 at 12 44 57 PM" src="https://user-images.githubusercontent.com/5309692/60209852-5be2ce00-9810-11e9-9d74-411043a2e22d.png">
<img width="604" alt="Ω 2019-06-26 at 12 45 16 PM" src="https://user-images.githubusercontent.com/5309692/60209857-60a78200-9810-11e9-91fc-c8546fe56ca6.png">

**Comment by khalwat** [comment]
Yep, I think that's a bug @lindseydiloreto -- or at least not what I'd expect either. Go ahead and change those, and explicitly set the transform type

But it's not going to change how it appears on Twitter unfortunately.

On Facebook, it'll look okay.

**Comment by lindseydiloreto** [comment]
Looks like that worked. Thanks @khalwat!

**Comment by khalwat** [comment]
That other issue is over here @lindseydiloreto https://github.com/nystudio107/craft-seomatic/issues/366

**Comment by circa42** [comment]
Looks like I have the same issue. No matter what I choose—specific to either the Twitter or FB image or in General—the image gets cropped. Since I'm working with book covers, that's a problem. I'd rather have them 'fit' with white space on left/right. But no matter what I chance, the image remains the same in the preview. 
![Screen Shot 2020-03-02 at 13 30 53]
<img width="1602" alt="Screen Shot 2020-03-02 at 13 32 13" src="https://user-images.githubusercontent.com/1124514/75719796-3ef97700-5c8a-11ea-8edd-448a3509443a.png">
(https://user-images.githubusercontent.com/1124514/75719708-1a9d9a80-5c8a-11ea-9d01-00d8c41519b0.png)

**Comment by circa42** [comment]
Expected result:
![expectedresult](https://user-images.githubusercontent.com/1124514/75818190-4c2b6a00-5d4d-11ea-96ef-bcdb17e06443.jpg)

**Comment by circa42** [comment]
Actual result: 
<img width="295" alt="actual" src="https://user-images.githubusercontent.com/1124514/75818302-7846eb00-5d4d-11ea-9bcc-eefe44bcc5c6.png">

**Comment by MattWilcox** [comment]
Ok so do we need to petition P&T for a "Contain" mode? I can not imagine any situation where "fit" would be desirable in this use case, but this is how it seems to work intentionally. Despite it being counter intuitive.
